src/search-manager/data/api.ts

```diff
@@ -243,36 +245,41 @@ export async function fetchSearchResults({
   });
 
   // Third query is to get the hits for collections, with all the filters applied.
-  queries.push({
-    indexUid: indexName,
-    q: searchKeywords,
-    filter: [
-      // top-level entries in the array are AND conditions and must all match
-      // Inner arrays are OR conditions, where only one needs to match.
-      collectionsFilter, // include only collections
-      ...extraFilterFormatted,
-      // We exclude the block type filter as collections are only of 1 type i.e. collection.
-      ...tagsFilterFormatted,
-    ],
-    attributesToHighlight: ['display_name', 'description'],
-    highlightPreTag: HIGHLIGHT_PRE_TAG,
-    highlightPostTag: HIGHLIGHT_POST_TAG,
-    attributesToCrop: ['description'],
-    cropLength: 15,
-    sort,
-    offset,
-    limit,
-  });
+  if (fetchCollections) {
+    queries.push({
+      indexUid: indexName,
+      q: searchKeywords,
+      filter: [
+        // top-level entries in the array are AND conditions and must all match
+        // Inner arrays are OR conditions, where only one needs to match.
+        collectionsFilter, // include only collections
+        ...extraFilterFormatted,
+        // We exclude the block type filter as collections are only of 1 type i.e. collection.
+        ...tagsFilterFormatted,
+      ],
+      attributesToHighlight: ['display_name', 'description'],
+      highlightPreTag: HIGHLIGHT_PRE_TAG,
+      highlightPostTag: HIGHLIGHT_POST_TAG,
+      attributesToCrop: ['description'],
+      cropLength: 15,
+      sort,
+      offset,
+      limit,
+    });
+  }
```

Comment:
Why are we fetching `collections` in a separate query? I had thought this whole time that it would be included in the main query ("First query is always to get the hits, with all the filters applied."). I'd really prefer that approach if possible - it's much more efficient, and it avoids having all this special-case code for collections.  Sorry for not spotting this in the earlier PR - I know it's already the case.

Reply:
Also, we don't use the 2nd "recently modified" query when retrieving components for a single collection either, so just one "fetch the collection components" query would be enough here.

Reply:
@bradenmacdonald Splitting up collections and components efficiently on frontend while maintaining pagination in the same query will be difficult or rather not worth the effort.

Reply:
@pomegranited Not sure if I understand but happy to leave this for #1103 :stuck_out_tongue_closed_eyes:

Reply:
@navinkarkera No worries, I'll make sure #1103 deals with this. :)

But am interested in @bradenmacdonald 's reply here about the pagination issue.

Reply:
Ah, I see. I guess I was thinking as if we'd already implemented https://github.com/openedx/frontend-app-authoring/issues/1230 . If you can do that quickly and it makes your life easier, go for it. Otherwise, we'll have to implement that post-MVP and clean this up. The home page is the only place where we have multiple searches happening, so removing that will be a really nice simplification.

I also forgot how <SearchContextProvider> works... I was thinking maybe it was something like:

```
const searchContext = useSearchContext();
const recentlyModified = useContentSearchResults(..., filter: "RECENT");
const components = useContentSearchResults(..., filter: "type = component");
const collections = useContentSearchResults(..., filter: "type = collection");
```
but it doesn't work that way - useSearchContext is really mostly designed to wrap a single search.

So again this can all be cleaned up nicely if we do https://github.com/openedx/frontend-app-authoring/issues/1230

Reply:
@bradenmacdonald Ohh, makes sense. I think handling it in a separate ticket would be better as this PR is already bigger than my expectations.